temp/OnnxTransformTempSolution/Microsoft.ML.Transforms.Onnx/OnnxUtils.cs

```diff
@@ -0,0 +1,209 @@
+            {
+                return new Tensor((System.Int16)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt32))
+            {
+                return new Tensor((System.UInt32)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt64))
+            {
+                return new Tensor((System.UInt64)(object)data);
+            }
+            else if (typeof(T) == typeof(System.UInt16))
+            {
+                return new Tensor((System.UInt16)(object)data);
+            }
+            throw new NotSupportedException($"Unsupported type {typeof(T)}");
+        }
+
+        /// <summary>
+        /// Sonoma API only provides Tensor() constructors with overloaded versions
+        /// based on data type. ML.NET cannot use the overloaded version and requires 
+        /// generic version. CreateTensor<T> is generic wrapper on top of
+        /// overloaded Tensor(T[] data, OnnxShape shape) constructors.
+        /// </summary>
+        public static Tensor CreateTensor<T>(T[] data, OnnxShape shape)
+        {
+            if (typeof(T) == typeof(System.Boolean))
+            {
+                return new Tensor(((System.Boolean[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Double))
+            {
+                return new Tensor(((System.Double[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Single))
+            {
+                return new Tensor(((System.Single[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Int32))
+            {
+                return new Tensor(((System.Int32[])(object)data).ToList(), shape);
+            }
+            else if (typeof(T) == typeof(System.Int64))
+            {
+                return new Tensor(((System.Int64[])(object)data).ToList(), shape);
+            }
+            throw new NotImplementedException($"Not implemented type {typeof(T)}");
+        }
+
+        /// <summary>
+        /// Sonoma API only provides CopyTo() functions with overloaded versions
+        /// based on data type. ML.NET cannot use the overloaded version and requires 
+        /// generic version. CopyTo<T> is generic wrapper on top of
+        /// overloaded Tensor.CopyTo(List<T> dst) methods.
+        /// Also Tensor.CopyTo(List<T> dst) requires a list input, whereas ML.NET
+        /// provides array buffers to copy values to. This mismatch causes an extra copy.
+        /// </summary>
+        public static void CopyTo<T>(Tensor tensor, T[] dst)
+        {
+            if (typeof(T) == typeof(System.Single))
+            {
+                // Sonoma only takes List<T>. We need to do an extra copy to T[]
+                var listDst = new List<System.Single>();
+                var typedDst = (System.Single[])(object)dst;
+                tensor.CopyTo(listDst);
+                listDst.CopyTo(typedDst);
```

Comment:
@Jignesh - Tensor.CopyTo(List<T> dst) requires a list input, whereas ML.NET provides array buffers to copy values to. This mismatch causes an extra copy. Do you know how to prevent this?

Reply:
@shmoradims . I'll add a CopyTo(T[] dst) method to Tensor, which should skip this extra copy.